src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -42,10 +46,60 @@ pub struct BusinessLogicSyscallHandler<T: State + StateReader> {
     pub(crate) block_info: BlockInfo,
     pub(crate) state: T,
     pub(crate) starknet_storage_state: ContractStorageState<T>,
+    pub(crate) internal_calls: Vec<CallInfo>,
+    pub(crate) expected_syscall_ptr: Relocatable,
 }
 
 impl<T: State + StateReader + Clone> BusinessLogicSyscallHandler<T> {
-    pub fn new(block_info: BlockInfo, contract_address: Address, state: T) -> Self {
+    pub fn new(
+        tx_execution_context: TransactionExecutionContext,
+        state: T,
+        resources_manager: ExecutionResourcesManager,
+        caller_address: Address,
+        contract_address: Address,
+        general_config: StarknetGeneralConfig,
+        syscall_ptr: Relocatable,
+    ) -> Self {
+        // TODO: check work arounds to pass block info
+        let block_info = BlockInfo::default();
+        let events = Vec::new();
+        let tx_execution_context = TransactionExecutionContext {
```

Comment:
This is confusing. I think the following would work better:
```suggestion
        let tx_execution_context = TransactionExecutionContext::default();
```

Reply:
you are right, fixed!